ReactiveCocoa/Swift/Signal.swift

```diff
@@ -334,10 +340,20 @@ public func materialize<T, E>(signal: Signal<T, E>) -> Signal<Event<T, E>, NoErr
 /// _values_ into a signal of those events themselves.
 public func dematerialize<T, E>(signal: Signal<Event<T, E>, NoError>) -> Signal<T, E> {
 	return Signal { observer in
-		return signal.observe(next: { event in
-			observer.put(event)
-		}, completed: {
-			sendCompleted(observer)
+		return signal.observe(SinkOf { event in
+			switch event {
+			case let .Next(innerEvent):
+				observer.put(innerEvent.unbox)
+
+			case .Error:
+				assert(false)
```

Comment:
This condition is literally impossible per the types—the compiler just requires it.